platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/BootstrapUtils.java

```diff
@@ -318,20 +318,39 @@ public static void writeSettingsUsed(@NonNull final Configuration configuration)
     /**
      * Determine which nodes should be run locally
      *
-     * @param addressBook       the address book
-     * @param localNodesToStart local nodes specified to start by the user
+     * @param addressBook   the address book
+     * @param cliNodesToRun nodes specified to start by the user on the command line
+     * @param envNodesToRun nodes specified to start by the user in the environment
      * @return the nodes to run locally
      */
     public static @NonNull List<NodeId> getNodesToRun(
-            @NonNull final AddressBook addressBook, @NonNull final Set<NodeId> localNodesToStart) {
+            @NonNull final AddressBook addressBook,
+            @NonNull final Set<NodeId> cliNodesToRun,
+            @NonNull final List<NodeId> envNodesToRun) {
         Objects.requireNonNull(addressBook);
-        Objects.requireNonNull(localNodesToStart);
+        Objects.requireNonNull(cliNodesToRun);
+        Objects.requireNonNull(envNodesToRun);
+
         final List<NodeId> nodesToRun = new ArrayList<>();
-        for (final Address address : addressBook) {
-            // if the local nodes to start are not specified, start all local nodes. Otherwise, start specified.
-            if (AddressBookNetworkUtils.isLocal(address)
-                    && (localNodesToStart.isEmpty() || localNodesToStart.contains(address.getNodeId()))) {
-                nodesToRun.add(address.getNodeId());
+        final Set<NodeId> addressBookNodeIds = addressBook.getNodeIdSet();
+
+        if (cliNodesToRun.isEmpty()) {
+            if (envNodesToRun.isEmpty()) {
+                // if no node ids are provided by cli or env, run all nodes from the address book.
+                return new ArrayList<>(addressBookNodeIds);
```

Comment:
This logic modification seems to require a change for all node operators (mainnet, testnet, all the perf/preview networks, and all testing frameworks that start a node via this code path), which is a lot of affected parties.

Is this being communicated to all the affected parties and configuration changes are being made to ensure that all the systems are configured before this code is released?

Reply:
Also, I don't see where ServicesMain would log an error if both the env vars and CLI args are missing. It only complains if more than one node is specified on the command line. I think you'd want to verify that logic.


If verified and/or fixed, so that the node indeed errors out when the config is totally missing everywhere, then this concern can be resolved. (Although you'd still want to communicate the need to update the configuration to all the parties, so that they don't get surprised with unexpected exceptions.)

Reply:
It's in the BootstrapUtils call after the code I modified.  https://github.com/hashgraph/hedera-services/blob/d0888fc879a8ff49089f41e80a4227a5bf568e0f/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/BootstrapUtils.java#L346-L356

Reply:
That looks good. Thanks! However, we still need to communicate the change to everyone who needs to update their nodes configurations before releasing this code.